lib/saml/user.rb

```diff
@@ -17,35 +17,37 @@ def last_signed_in
       Time.current.utc
     end
 
-    def loa
-      { current: loa_current, highest: loa_highest(attributes) }
-    end
-
     def to_hash
       Hash[serializable_attributes.map { |k| [k, @decorated.send(k)] }]
     end
 
+    def authn_context
```

Comment:
Nitpick - I think this could stand to be renamed, as it's overloading the term somewhat with how it's used in session_controller. E.g. now in the multifactor method when building the SAML request you set `connect` to the value of this authn_context, and `authn_context` to `multifactor`. A different name for this method like "auth flavor" or something along those lines might make this less confusing for the next person.

Reply:
But then you would have to rename it on user, these attributes get deserialized into current_user, and they also appear in the user_serializer. It's not really overloaded, it's precisely from where the F/E gets its authn_context. The only place where it is different is in saml_settings, and even there it kind of stands to reason that we could use the value from user (at least for the ones that have a session).

Let me think about this a bit more, to me it's not really overloaded, it's precisely the same thing, but originates from two different sources.